src/crimson/osd/backfill_state.h

```diff
@@ -143,13 +146,25 @@ struct BackfillState {
     sc::result react(const Triggered&);
   };
 
+  struct Cancelled : sc::state<Cancelled, BackfillMachine>,
+                     StateHelper<Cancelled> {
+    using reactions = boost::mpl::list<
+      sc::custom_reaction<Triggered>,
+      sc::transition<sc::event_base, Crashed>>;
```

Comment:
I think `PrimaryScanned`, `ReplicaScanned` and `ObjectPushed` should be treated as valid events, am I right?

Reply:
@rzarzynski Could you take a look at this comment?

Reply:
@xxhdx1985126: I understood your question as _can e.g. `ReplicaScanned` event happen while the machine is in the `Cancelled`_. My very first thought is they can and they should ignored, I guess.

Reply:
> @xxhdx1985126: I understood your question as _can e.g. `ReplicaScanned` event happen while the machine is in the `Cancelled`_. My very first thought is they can and they should ignored, I guess.

Ah, yes, thanks. But this brings me to another question, what if those old events arrives when a new round of backfill has started? I also recorded this question in the comment: https://github.com/ceph/ceph/blob/44c4a176d6fc8641c00bf111e4e8f647dda36779/src/crimson/osd/pg_recovery.cc#L636-L640